### PR TITLE
CDAP-12633 Add error datasets within a transaction.

### PR DESCRIPTION
**Pull request by anwar6953**
https://issues.cask.co/browse/CDAP-12633

https://builds.cask.co/browse/CDAP-RUT1401-2
Integration tests against this branch: https://builds.cask.co/browse/CDAP-ITM19-35

### CODE REVIEW

File: cdap-app-templates/cdap-etl/cdap-etl-batch/src/main/java/co/cask/cdap/etl/batch/mapreduce/ETLMapReduce.java [hunk 1]
```diff
@@ -356,17 +356,12 @@ public void act(DefaultJoinerContext joinerContext) {
...
-    // setup time partition for each error dataset
-    for (StageSpec stageInfo : Sets.union(phase.getStagesOfType(Transform.PLUGIN_TYPE),
-                                          phase.getStagesOfType(BatchSink.PLUGIN_TYPE))) {
-      if (stageInfo.getErrorDatasetName() != null) {
-        Map<String, String> args = new HashMap<>();
-        args.put(FileSetProperties.OUTPUT_PROPERTIES_PREFIX + "avro.schema.output.key",
-                 Constants.ERROR_SCHEMA.toString());
-        TimePartitionedFileSetArguments.setOutputPartitionTime(args, context.getLogicalStartTime());
-        context.addOutput(Output.ofDataset(stageInfo.getErrorDatasetName(), args));
+    context.execute(new TxRunnable() {
```
Comment (albertshau) [inline cdap-app-templates/cdap-etl/cdap-etl-batch/src/main/java/co/cask/cdap/etl/batch/mapreduce/ETLMapReduce.java]: if we don't want this to be required, should put a comment here with a link to a relevant jira.

**Review by albertshau — COMMENTED** [review]
lgtm

File: cdap-app-templates/cdap-etl/cdap-etl-batch/src/main/java/co/cask/cdap/etl/batch/mapreduce/ETLMapReduce.java (hunk 1 above)
Comment (anwar6953) [reply cdap-app-templates/cdap-etl/cdap-etl-batch/src/main/java/co/cask/cdap/etl/batch/mapreduce/ETLMapReduce.java]: Created a [JIRA](https://issues.cask.co/browse/CDAP-12634) and updated this line to be preceded with a comment.

**Comment by anwar6953** [comment]
Integration tests and unit tests passing, so merged.

**Pull request merged**